### ISSUE TITLE
current client and tusted mac

### ISSUE THREAD

**Issue by laoshaw** [openNDS/openNDS#143]
I ran `ndsctl status` or `ndsctl json` and saw multiple current clients, from the MAC and IP only one made sense to me, also I modified opennds to require username/password so it's unlikely some nearby users just logged in, where are those 'current client' from? will ndsctl show any attempted user logins as active clients even though they failed to login?

additionally if I add a trusted-mac, it has nothing in the log file to show a device is let in, can logfile contain that piece of info as well, login.sh might not be able to that but adding that info to C will also be helpful.

basically I need a way to show real current live clients, be it MAC-trusted or logged in.

**Comment by bluewavenet** [comment]
@laoshaw 
ndsctl status and json both show all clients that have triggered openNDS via CPD. Some may be authenticated and some may be preauthenticated. Preauthenticated means they have triggered openNDS but have not been authenticated.

Check again, you will find the leaching clients will be preauthenticated if you require a secret username/password.
In contrast, the logs only show clients that have been authenticated.
Trusted clients are just that - they are trusted to do anything. You can see "real current live clients" by looking at the contents of /tmp/dhcp.leases. You can cross check by running `ip neigh`

**A trusted client will not trigger openNDS as it has Internet access through the firewall so there is no way for openNDS to add it to logs or anything else.**

The question you should ask yourself is: _why do I have trusted clients?_ A trusted device eg an IoT device that has to "call home" is self explanatory. A user's device is better handled by the captive portal. Special users can be assigned unlimited access so they only have to login once (or at least until the next restart of openNDS). You could even give login.sh a list of mac addresses to which it serves just a click to continue, whereas mac addresses not on the list have to log in.

**Comment by bluewavenet** [comment]
@laoshaw 
The config option "preauthidletimeout" is the interval in minutes at which openNDS polls for live preauthenticated clients, deleting them if idle.
I think the default is 30 minutes - so any of your neighbours blocked from access by your password will remain in the preauthenticated user list for a maximum time of that interval.
You can reduce preauthidletimeout to clear out the unwanted clients earlier, but if you make this too small it can create difficulties for genuine clients if they do not log in quick enough.
